packages/vscode-extension/src/notifications/AlertsHandler.ts

```
@@ -0,0 +1,66 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Notification, NotificationSeverity } from "@kogito-tooling/notifications/dist/core";
+import { WorkspaceApi } from "@kogito-tooling/workspace/dist/api";
+import * as vscode from "vscode";
+import { NotificationHandler } from "./NotificationHandler";
+
+class AlertsHandler implements NotificationHandler {
+  constructor(private readonly workspaceApi: WorkspaceApi) {}
+  send(notification: Notification): void {
+    this.get(notification.severity)(notification.message, notification.path);
+  }
+  set(path: string, notifications: Notification[]): void {
+    const errors = notifications.filter(n => n.severity === NotificationSeverity.ERROR);
+    const others = notifications.filter(n => n.severity !== NotificationSeverity.ERROR);
+
+    const errorsMessage = this.consolidateMessages(errors);
+    const othersMessage = this.consolidateMessages(others);
+
+    this.get(NotificationSeverity.ERROR)(errorsMessage, path);
+    this.get(NotificationSeverity.SUCCESS)(othersMessage, path);
+  }
+  remove(path?: string): void {}
+
+  get(severity: string) {
+    if (severity == NotificationSeverity.ERROR) {
+      return this.handleStrategy(vscode.window.showErrorMessage);
+    } else if (severity == NotificationSeverity.WARNING) {
+      return this.handleStrategy(vscode.window.showWarningMessage);
+    } else {
+      return this.handleStrategy(vscode.window.showInformationMessage);
```

```suggestion
    }
    return this.handleStrategy(vscode.window.showInformationMessage);
```

---

Actually, I think a switch/case would be more recommended here. We're making "INFO" the default severity, but that's not what we want. We want severity to be EXPLICITLY informed every time. We should make use of the `default` clause and throw an exception whenever a message comes with an "unknown" severity. That can happen if a new severity type is created and this method is not properly updated. Having an exception be thrown on a default clause will potentially prevent future bugs.

---

Agreed!